src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -147,38 +144,54 @@ public String produceDts(Compiler compiler) {
         int lastDot = symbol.getName().lastIndexOf('.');
         namespace = lastDot >= 0 ? symbol.getName().substring(0, lastDot) : "";
       }
-      emitNoSpace("declare namespace ");
-      emitNoSpace(INTERNAL_NAMESPACE);
-      if (!namespace.isEmpty()) {
-        emitNoSpace("." + namespace);
-      }
-      emitNoSpace(" {");
-      indent();
-      emitBreak();
-      if (symbol.getType().isEnumType() || symbol.getType().isFunctionType() ||
-          !symbol.getType().isObject()) {
-        walkScope(symbol, namespace, isDefault);
-      } else {
-        // JSCompiler treats "foo.x" as one variable name, so collect all provides that start with
-        // $provide + ".".
-        String prefix = provide + ".";
-        for (TypedVar other : topScope.getAllSymbols()) {
-          String otherName = other.getName();
-          if (otherName.startsWith(prefix) && other.getType() != null
-              && !other.getType().isFunctionPrototypeType()) {
-            walkScope(other, namespace, false);
-          }
-        }
-      }
-      unindent();
-      emit("}");
-      emitBreak();
+      declareNamespace(namespace, provide, symbol, isDefault, allSymbols);
       declareModule(provide, isDefault);
     }
     checkState(indent == 0, "indent must be zero after printing, but is %s", indent);
     return out.toString();
   }
 
+  private void declareNamespace(String namespace, String provide, TypedVar symbol,
+      boolean isDefault, Iterable<TypedVar> allSymbols) {
+    emitNoSpace("declare namespace ");
+    emitNoSpace(INTERNAL_NAMESPACE);
+    if (!namespace.isEmpty()) {
+      emitNoSpace("." + namespace);
+    }
+    emitNoSpace(" {");
+    indent();
+    emitBreak();
+    if (symbol.getType().isEnumType() || symbol.getType().isFunctionType() ||
```

Comment:
I can't convince myself why this can't just be if (isDefault)

isDefault = `!symbol.getType().isObject() || symbol.getType().isInterface() || symbol.getType().isFunctionType();`
checkHere = `symbol.getType().isEnumType() || symbol.getType().isFunctionType() ||!symbol.getType().isObject()`

so the symmetric diff is `isEnumType` and `isInterface`. Why don't threat enums as default exports? And why would we go into the `else` branch here if the type is interface?